### PR TITLE
Add Google Analytics, Matoma/Piwik Tracking #2612

### LINKED ISSUE #2612: Admin: Add options for Matomo and Google Analytics
Currently we are collecting site info using atomo.anomic.de/piwik.js:1. Please implement config options for analytics tools in the admin section. So, admin users can add the Google or Matomo key. Open an issue and PR on the server to store the info accordingly.

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2612

- [x] Changes: Added Google Analytics Tracking
- [x] Matomo already added
- [x] Fetch keys from backend
- [x] scrollToTop on route changed

Demo Link: https://pr-2634-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by mariobehling** [comment]
I am unclear about this PR. Please store the info on the server. I think the issue is already clear about the goals. I don't see it implemented here. Did you read the issue description?

**Comment by fragm3** [comment]
@mariobehling on it

**Comment by fragm3** [comment]
@mariobehling @akshatnitd implemented Google Analytics for tracking react. Using server-side key for getting tracking ID and tracking would be disabled if the key is null, i.e if the field is not present
Screenshot 2019-07-19 at 11 18 35 PM

**Comment by mariobehling** [comment]
Yes, good. Thanks. Please also implement Matamo like that.

**Comment by fragm3** [comment]
@akshatnitd @mariobehling I have added piwik/matoma, using library https://www.npmjs.com/package/react-piwik. 

I don't understand the use case for making this server-side, the [doc](https://www.npmjs.com/package/react-piwik) say to put it into the entry point which means the values should be hardcoded.  Talking about ```url: 'your-piwik-server-url.com',
  siteId: 12,``` part

**Comment by mariobehling** [comment]
@fragm3 In the page you linked I see custom fields e.g. url: 'your-piwik-server-url.com'. The goal is to be able to control these custom form elements on the admin UI. So, for Google you have to add a analytics string and for Matoma the custom server (where Matoma is deployed) and ID I guess. Anything else that I am missing?

**Comment by fragm3** [comment]
@mariobehling @akshatnitd I have implemented matomo, google analytics, both are taking keys from backend now.

**Comment by fragm3** [comment]
Screenshot 2019-07-20 at 11 11 01 PM

File: src/withTracker.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      page,
+      ...options,
+    });
+    GoogleAnalytics.pageview(page);
+  };
+
+  // eslint-disable-next-line
+  const HOC = class extends Component {
+    componentDidMount() {
+      scrollAnimation(document.documentElement, 0, 150, 'vertical');
```
Comment (fragm3) [inline src/withTracker.js]: Added scrollToTop on route changed @akshatnitd

**Pull request merged**